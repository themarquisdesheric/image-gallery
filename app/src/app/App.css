.App {
  text-align: center;
}

.App-header {
  text-align: center;
  background-color: #222;
  height: 40px;
  padding: 20px;
  margin: 0;
  color: white;
}

.App-header span {
  margin: 0 20px;
}

.Nav-list {
  background-color: #fff;
  margin: 0;
  padding: .5em;
}

.Nav-list li {
  display: inline;
  padding: 1em;
}

.selected {
  font-weight: bold;
}

ul {
  list-style-type: none;
  padding-left: 0;
}

li {
  padding: .1em 0;
}

.button-container {
  padding: .5em 0;
}

.Add {
  margin: 1em auto 0;
  padding-top: 1em;
  width: 60%;
}

.Add-header {
  padding: 0 .5em;
}

.Add-icon {
  margin-bottom: 5px;
  margin-right: 5px;
}

fieldset {
  padding-bottom: 1em;
}

form {
  margin-bottom: 1em;
}

input {
  margin: 0 1em;
}

.Thumbnail {
  display: inline-block;
  height: 185px;
  width: 150px;
  overflow: hidden;
}

.Thumbnail-image {
  height: 125px;
  width: 125px;
  border: 1px solid #222;
}

.Gallery-image {
  max-width: 50%;
  border: 1px solid #222;
}

.out-of-bunnies {
  color: #f00;
}

.list-title {
  font-weight: bold;
}

.Album-list-delete-button {
  margin-left: 10px;
}

.delete-button {
  margin-top: 1em;
}