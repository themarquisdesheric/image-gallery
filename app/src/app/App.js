import React, { Component } from 'react';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import AppHeader from '../components/AppHeader';
import Home from '../components/Home';
import Albums from '../components/Albums';
import './App.css';

class App extends Component {
  render() {
    return (
      <Router>
        <div className="App">
          <AppHeader />
          <Route exact path="/" component={Home} />
          <Route path="/albums" component={Albums} />
        </div>
      </Router>
    );
  }
}

export default App;
