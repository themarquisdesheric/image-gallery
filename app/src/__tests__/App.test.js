import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import App from '../app/App';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
});

it('renders a snapshot', () => {
  const tree = renderer.create(<App/>).toJSON();
  expect(tree).toMatchSnapshot();
});