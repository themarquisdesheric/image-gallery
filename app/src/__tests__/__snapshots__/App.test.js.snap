// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
<div
  className="App"
>
  <div>
    <h1
      className="App-header"
    >
      <span
        aria-label="bunny"
        role="img"
      >
        üê∞
      </span>
      Image Gallery
      <span
        aria-label="bunny"
        role="img"
      >
        üê∞
      </span>
    </h1>
    <nav>
      <ul
        className="Nav-list"
      >
        <li>
          <a
            className="selected"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            Home
          </a>
        </li>
        <li>
          <a
            className={undefined}
            href="/albums"
            onClick={[Function]}
            style={undefined}
          >
            Albums
          </a>
        </li>
      </ul>
    </nav>
  </div>
  <div>
    <h2>
      Home
    </h2>
    <p>
      Welcome to the Image Gallery!
    </p>
  </div>
</div>
`;
